lib/output/generatePage.js

```diff
@@ -15,13 +15,13 @@ var callPageHook = require('./callPageHook');
     @param {Page} page
     @return {Promise<Page>}
 */
-function generatePage(output, page) {
+function generatePage(output, page, pagePath) {
```

Comment:
I don't think this is a good idea to pass the `pagePath` as a second argument to all functions, we can change the data structure for `Page` to more suitable to this new feature instead.